pkg/sloop/store/untyped/store.go

```diff
@@ -16,21 +16,45 @@ import (
 	"time"
 )
 
-func OpenStore(factory badgerwrap.Factory, rootPath string, configPartitionDuration time.Duration) (badgerwrap.DB, error) {
+func OpenStore(factory badgerwrap.Factory, rootPath string, configPartitionDuration time.Duration, badgerMaxTableSize int64, badgerKeepL0InMemory bool, badgerVLogFileSize int64, badgerVLogMaxEntries uint, badgerUseLSMOnlyOptions bool) (badgerwrap.DB, error) {
+	if configPartitionDuration != time.Hour && configPartitionDuration != 24*time.Hour {
+		return nil, fmt.Errorf("Only hour and day partitionDurations are supported")
+	}
+
 	err := os.MkdirAll(rootPath, 0755)
 	if err != nil {
 		glog.Infof("mkdir failed with %v", err)
 	}
 	// For now using a temp name because this all need to be replaced when we add real table/partition support
-	opts := badger.DefaultOptions(rootPath)
+	var opts badger.Options
+	if badgerUseLSMOnlyOptions {
+		opts = badger.LSMOnlyOptions(rootPath)
+	} else {
+		opts = badger.DefaultOptions(rootPath)
+	}
+
+	if badgerMaxTableSize != 0 {
+		opts = opts.WithMaxTableSize(badgerMaxTableSize)
+	}
+	opts.KeepL0InMemory = badgerKeepL0InMemory
+	if badgerVLogFileSize != 0 {
+		opts = opts.WithValueLogFileSize(badgerVLogFileSize)
+	}
+	if badgerVLogMaxEntries != 0 {
+		opts = opts.WithValueLogMaxEntries(uint32(badgerVLogMaxEntries))
+	}
 
 	db, err := factory.Open(opts)
 	if err != nil {
 		return nil, fmt.Errorf("badger.OpenStore failed with: %v", err)
 	}
 
-	if configPartitionDuration != time.Hour && configPartitionDuration != 24*time.Hour {
-		return nil, fmt.Errorf("Only hour and day partitionDurations are supported")
+	glog.Infof("BadgerDB Options: %+v", opts)
+	lsm, vlog := db.Size()
+	glog.Infof("BadgerDB Size lsm=%v, vlog=%v", lsm, vlog)
```

Comment:
wondering if this logging should be moved to a separate method and called here as well as before kicking off the GC?

Reply:
Done